app/controllers/TaxAccountSummaryController.scala

```diff
@@ -52,17 +52,12 @@ trait TaxAccountSummaryController extends TaiBaseController
           ServiceCheckLite.personDetailsCheck {
 
             val nino = Nino(user.getNino)
-            val taxSummaryFuture = taxAccountService.taxAccountSummary(nino, TaxYear())
-            val taxCodeIncomesFuture = taxAccountService.taxCodeIncomes(nino, TaxYear())
-            val nonTaxCodeIncomeFuture = taxAccountService.nonTaxCodeIncomes(nino, TaxYear())
             auditService.createAndSendAuditEvent(TaxAccountSummary_UserEntersSummaryPage, Map("nino" -> user.getNino))
-            val employmentsFuture = employmentService.employments(nino, TaxYear())
-
             for {
-              taxSummary <- taxSummaryFuture
-              taxCodeIncomes <- taxCodeIncomesFuture
-              nonTaxCodeIncome <- nonTaxCodeIncomeFuture
-              employments <- employmentsFuture
+              taxSummary <- taxAccountService.taxAccountSummary(nino, TaxYear())
+              taxCodeIncomes <- taxAccountService.taxCodeIncomes(nino, TaxYear())
+              nonTaxCodeIncome <- taxAccountService.nonTaxCodeIncomes(nino, TaxYear())
+              employments <- employmentService.employments(nino, TaxYear())
```

Comment:
Is it possible to prevent employments and isAnyFormInProgress  calls
Can we refactor the match statement in yield clause as employments and isAnyFormInProgress is never compared?